lib/twingly/url.rb

```diff
@@ -101,7 +99,7 @@ def normalized
       normalized_url.host   = normalized_host
       normalized_url.path   = normalized_path
 
-      self.class.internal_parse(normalized_url)
+      self.class.parse(normalized_url)
```

Comment:
I think it's pretty easily amended by splitting `internal_parse` into two different ones, one for addressable and one for public suffix